fv/infrastructure/infra_k8s.go

```diff
@@ -400,22 +411,37 @@ func (kds *K8sDatastoreInfra) Stop() {
 	kds.needsCleanup = true
 }
 
+type cleanupFunc func(clientset *kubernetes.Clientset, calicoClient client.Interface)
+
 func (kds *K8sDatastoreInfra) CleanUp() {
 	log.Info("Cleaning up kubernetes datastore")
-	cleanupAllPods(kds.K8sClient)
-	cleanupAllNodes(kds.K8sClient)
-	cleanupAllNamespaces(kds.K8sClient)
-	cleanupAllPools(kds.calicoClient)
-	cleanupIPAM(kds.calicoClient)
-	cleanupAllGlobalNetworkPolicies(kds.calicoClient)
-	cleanupAllNetworkPolicies(kds.calicoClient)
-	cleanupAllHostEndpoints(kds.calicoClient)
-	cleanupAllFelixConfigurations(kds.calicoClient)
-	cleanupAllServices(kds.K8sClient)
+	startTime := time.Now()
+
+	var wg sync.WaitGroup
+	for _, f := range []cleanupFunc{
+		cleanupAllPods,
+		cleanupAllNodes,
+		cleanupAllNamespaces,
```

Comment:
I wonder if the order completely does not matter. Can you clean up namespace before you clean up pods in it? Or does it get blocked until the other colliding thing completes?

Reply:
I guess there may be *some* blocking, but we still benefit by parallelising for cleanups that are independent.

Reply:
Yes, deleting a namespace deletes all pods inside by the garbage collector.  But that's slower.